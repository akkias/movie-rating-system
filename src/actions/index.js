export function getMovies(movies) {
    return {
        type: "GET_MOVIES",
        movies
    }
}