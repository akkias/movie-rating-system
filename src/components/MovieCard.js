import React from 'react';
import {sortDesc} from '../utils/functionUtils'
import Rating from 'react-rating';

const renderMovie = (props) => {
    return(
            props.movies && sortDesc(props.movies).map((movie, index) => {
                return(
                    <div className="movie-card" key={movie.id}>
                        <div className="movie-card-poster">
                            <img alt={movie.title} src={`/assets/images/posters/${movie.poster}`} />
                        </div>
                        <div className="movie-card-footer">
                            <h4 className="movie-card-title">{movie.title}</h4>
                            <Rating initialRating={movie.rating}
                                emptySymbol="far fa-star"
                                fullSymbol="fas fa-star"
                                className="movie-card-rating"
                                onClick={(rating) => props.setRatingFunc(rating, index, movie.id)}
                            />
                        </div>
                    </div>
                )
            }
        )
    )
}

const MovieCard = (props) => {  
    return(
        <div className="movie-card-container">
            {(props.movies && props.movies.length >= 1) && renderMovie(props)}
        </div>
    )
}
export default MovieCard;