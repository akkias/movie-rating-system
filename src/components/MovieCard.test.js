import React from 'react';
import { mount } from 'enzyme';
import MovieCard from './MovieCard'
import {setRating} from '../actions';
const fetchMock = require('fetch-mock');



describe('<MovieCard />', () => {

    const props = {'movies': [{
                        "id": 1,
                        "title": "Mission: Impossible - Fallout",
                        "rating": 4,
                        "poster": "mi.jpg"
                        }]
                    }
    const movieCard = mount(<MovieCard {...props} />);
    it('<movieCard /> Renders correctly', () => {
        expect(<movieCard />).toMatchSnapshot();
    });

    it('fires the function to rate the movie', () => {
        const expectedAction = {type: 'SET_RATING', payload:{id:1,index:2,rating:3}};
        expect(setRating(3,2,1)).toEqual(expectedAction);
    })

    fetchMock.restore();
})
