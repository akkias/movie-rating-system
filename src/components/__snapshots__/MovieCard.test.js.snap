// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieCard /> <movieCard /> Renders correctly 1`] = `<movieCard />`;
